libosmscout-client-qt/src/osmscout/MapWidget.cpp

```diff
@@ -277,7 +277,7 @@ void MapWidget::paint(QPainter *painter)
       }
 
       painter->save();
-      QTransform t=QTransform::fromTranslate(x, y); // move to rotation center
+      QTransform t=QTransform::fromTranslate(x * pixelRatio, y * pixelRatio); // move to rotation center
```

Comment:
I don't understand where is the problem here (always have problem with these transformation matrices :-) ). Why map icons, that are SVG images too, are rendered on correct place but vehicle icon is on the wrong place? Or map tiles during tiled rendering where Transformation is used too?

May it be caused by the fact that icon and painter have different device pixel ratio? Can you try to add these debug lines please?
```
qDebug() << "icon pixel ratio: " << vehicleIcon.devicePixelRatio();
qDebug() << "painter pixel ratio: " << painter->device()->devicePixelRatioF();
```

Reply:
both are 1.0 on my desktop...

Reply:
Not the correct place and not the correct size. In my sample the screen pixel ratio is 3.

Reply:
I try the debug output on macos with retina screen which has a HiDPI ratio of 2...

Reply:
Debug output shows:
```
icon pixel ratio: 1
painter pixel ratio: 1
```

Reply:
But other usages of `projection.GeoToPixel` have to be wrong too, or not? Where is the translation between map painter coordinates and upscaled screen coordinates? How it is possible that tile rendering on rotated map is correct?! Did you understand my concerns?

Reply:
I try to understand ;).
I your desktop is Linux/Gnome i.e ubuntu, try to upscale the screen with the factor 200% and you will see what happens... the HiDPI will be 2.0

Reply:
Also the tile rendering is correct

Reply:
How do you make your test ? Have you an app for desktop I can test here with the HiDPI ?

Reply:
If your desktop is Linux, you could try with my app. I guess you have Qt headers installed. To build it, clone the branch dev/20210323:
```
git clone -b dev/20210323  https://github.com/janbar/osmin.git
cd osmin
git submodule init
git submodule update
mkdir build
cd build
cmake -B build .
cmake --build build/ -j8
```
finally launch the app: ./build/osmin
Download a map or copy an existing map into the folder $HOME/Maps/
In the file `src/osmin.cpp` line 330 you can set a fixed value for ScreenScaleFactor (i.e QVariant(1.0))
Then scale your screen in the Display/Parameters to 200%, restart the app ...

Reply:
I was able to reproduce it with your procedure and NavigationSimulation demo. It is really strange. When I replaced usage of QTransform with these two lines:
```
      painter->translate(x,y);
      painter->rotate(iconAngle.AsDegrees());
```
it is working perfectly... I would guess that there is some bug in Qt or I just don't understand QTransform :-)

So, I would advice you to replace transform in your branch and rename `pixelRatio` to "vehicle icon scale" or something like this :-)